system-addon/content-src/components/TopSites/TopSite.jsx

```diff
@@ -21,25 +21,38 @@ export class TopSiteLink extends React.PureComponent {
    */
   _allowDrop(e, index) {
     let draggedIndex = parseInt(e.dataTransfer.getData("text/topsite-index"), 10);
-    if (!isNaN(draggedIndex) && draggedIndex !== index) {
+    if (!isNaN(draggedIndex)) {
       return true;
     }
     return false;
   }
+
+  /*
+   * Helper to determine whether the drop zone should allow a dragEnter. This is to
+   * avoid entering the site that is already being dragged during "drop preview".
+   */
+  _allowEnter(e, index, url) {
+    return this._allowDrop(e, index) && (!url || e.dataTransfer.getData("text/topsite-url") !== url);
```

Comment:
I don't quite understand why this is necessary? If I just have "dragenter" do the same thing as "dragover" and "drop" in the `switch` statement (i.e., just an `_allowDrop`), things seem to behave the same ??

Reply:
Without this check, the top sites would infinitely flicker between the preview and the original state. I'll verify this is still needed as well.

Reply:
Oh hey. I removed it and nothing broke. I guess it was a red herring.